### ISSUE TITLE
Memory leak with CPLEX DLL

### ISSUE THREAD

**Issue by GoogleCodeExporter** [coin-or/pulp#28]
```
What steps will reproduce the problem?
1. Solve a lot of lp problems using CPLEX_DLL (like the attached one)
2. Watch the memory consumption of your process growing up

What is the expected output? What do you see instead?

A constant memory consumption, like the one with - by example - CPLEX_CMD

What version of the product are you using? On what operating system?

1.4.7 as well as the trunk, on a Cent OS (GNU/linux 32bits) computer, as well 
on other unix computers.

Please provide any additional information below.

The joined example is a stupid lp problem, but repeated a lot of time. The 
memory usage is growing up (as reported by (h)top), until there is no memory 
left and cplex refuse to solve another problem.

I've tried to fix it myself, without any success. The way I tried are : free 
the memory used by CPLEX by calling CPXfreeprob in CPLEX_DLL.__del__ (according 
to CPLEX doc, this must be done before releasing the license), and make the lib 
attribute (library loaded by ctypes) an attribute of instances of CPLEX_DLL 
instead of class attributes, to get the lib object deleted on instance delete.

(Anyway, thanks for pulp, it's a really good ilp API!)
```

Original issue reported on code.google.com by `flor...@biree.name` on 27 Jul 2010 at 9:37

Attachments:
- [memtest.py](https://storage.googleapis.com/google-code-attachments/pulp-or/issue-28/comment-0/memtest.py)

**Comment by pchtsp** [comment]
I'm closing this because of the new issue https://github.com/coin-or/pulp/issues/314
